azure-kusto-ingest/azure/kusto/ingest/ingestion_properties.py

```diff
@@ -107,21 +160,43 @@ def __init__(
         if storage_data_type:
             self.properties[self.STORAGE_DATA_TYPE] = storage_data_type
 
-    def is_valid(self, kind: IngestionMappingKind) -> bool:
+    def get_validation_errors(self, kind: IngestionMappingKind) -> List[str]:
         if not self.column:
-            return False
-
-        if kind in (IngestionMappingKind.JSON, IngestionMappingKind.PARQUET, IngestionMappingKind.ORC, IngestionMappingKind.W3CLOGFILE):
-            return (
-                bool(self.properties.get(self.PATH))
-                or self.properties.get(self.TRANSFORMATION_METHOD) == TransformationMethod.SOURCE_LINE_NUMBER.value
-                or self.properties.get(self.TRANSFORMATION_METHOD) == TransformationMethod.SOURCE_LOCATION.value
-            )
+            return ["Column name is required"]
+
+        results = []
+
+        for (prop, valid_types) in self.VALID_MAPPINGS.items():
+            if prop not in self.properties:
+                continue
+            if kind not in valid_types:
+                results.append(f"{prop} is not a valid mapping property for {kind}")
+
+        needed_props = self.NEEDED_PROPERTIES.get(kind, [])
+        if not needed_props:
+            results.append(f"{kind} is not a valid mapping kind")
```

Comment:
This is a strange message - WDYM it's not a valid kind?

Reply:
It's not a key of the map, so it's not in IngestionMappingKind

If the user gives something else, it's a better error than the generic ones.

Reply:
Ok removed